navjson/navjson.go

```diff
@@ -0,0 +1,58 @@
+// Package navjson provides a `nav.json` parser and validator.
+package navjson
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// NavFile represends a file/path association in a `nav.json` document.
+type NavFile struct {
+	Path  string    `json:"path"`
+	Files []NavFile `json:"files,omitempty"`
+}
+
+// NavPackage represents a list of packages in a `nav.json` document. This is only used for monorepos.
+type NavPackage map[string]string
+
+// NavJSON represets a `nav.json` document.
+type NavJSON struct {
+	Root             string     `json:"root"`
+	AssetRoot        string     `json:"assetRoot"`
+	SkipMenuOrdering bool       `json:"skipMenuOrdering,omitempty"`
+	Packages         NavPackage `json:"packages,omitempty"`
+	Files            []NavFile  `json:"files"`
+}
+
+// Parse attempts to parse the given data into a NavJSON object.
+func Parse(data string) (NavJSON, error) {
+	j := NavJSON{}
+	b := []byte(data)
+	err := json.Unmarshal(b, &j)
+	if err != nil {
+		return j, err
+	}
+
```

Comment:
I think we should expand on the `typing` of `Packages` and ensure that when it exists it is `key-value` pair of strings.

Also, should we ensure `skipMenuOrdering` when it exists is bool?

See - https://github.com/dequelabs/attest-node-suite/blob/develop/docs/nav.json as reference

Reply:
> Also, should we ensure skipMenuOrdering when it exists is bool?

The JSON parsing will fail if it's of the wrong type. I'll add a test showing this.